### DISCUSSION TITLE
Command for Processing Centroid Coordinates

### DISCUSSION THREAD

**Discussion by JadeUoM** [neperfepx/neper#854] [Q&A] [answered]
Hi Quey,

Thank you for the powerful platform you've created. 

I'm currently working with centroid coordinates of grains, but I'm a bit confused about the difference between "seed" and "centroid".   Could you explain the difference between them?

Additionally, if I have a file with the centroid coordinates, which command should I use? At the moment, I have two commands that I'm considering, but I want to ensure I'm using the correct one, which are:

neper -T -domain "cube(10, 10, 10)" -n "from_morpho"  **-morpho "centroid:file(Centroid.txt)"** -format "tess,ori,stl" -o Case1
&
neper -T -domain "cube(10, 10, 10)" -n "15" **-morphooptiini "coo:file(Centroid.txt),weight:0"** -format "tess,ori,stl" -o Case2

Could you please let me know which command is appropriate for processing the centroids coordinates, or if you have any better suggestions?? Any guidance would be greatly appreciated.

Thanks in advance for your help!

Best regards,
Jade

**Comment by rquey** [comment, marked as answer]
A _seed_ is a point from which a cell is generated. A _centroid_ is the geometrical center a the cell. They don't coincide.

The command
```
$ neper -T -domain "cube(10, 10, 10)" -n "from_morpho" -morpho "centroid:file(Centroid.txt)" -format "tess,ori,stl" -o Case1
```
seeks for the seed parameters such that the cell centroids match the specified centroids.

The command
```
$  neper -T -domain "cube(10, 10, 10)" -n "15" -morphooptiini "coo:file(Centroid.txt),weight:0" -format "tess,ori,stl" -o Case2
```
uses the cell centroids as seed coordinates.  The two may be close but do not exactly match.

Approach 2 is the standard way of doing it and give approximate results. Approach 1 will give exact results. Note that it will play with the cell sizes as well (compared to approach 2), using the seed weights. There might be more than one solutions to your problem, so, keep an eye out for possible oddities.

**Reply by JadeUoM** [reply]
Many thanks, Quey.
